services/analytics-service.ts

```diff
@@ -21,13 +21,16 @@ export class AnalyticsService implements IAnalyticsService {
 		private $prompter: IPrompter,
 		private $userSettingsService: UserSettings.IUserSettingsService,
 		private $analyticsSettingsService: IAnalyticsSettingsService,
-		private $options: IOptions) { }
+		private $options: IOptions) {
+			this.settingsNames = [this.$staticConfig.TRACK_FEATURE_USAGE_SETTING_NAME, this.$staticConfig.ERROR_REPORT_SETTING_NAME];
+			this.settingsNames.forEach(settingName => this.getStatus(settingName).wait());
```

Comment:
This should not be here as it is causing cyclic dependencies error in some cases